i18n: Add French translation

Kirwiisp opened the pull request:
Hi ! 
This is just to add a first version of a french translation !

Kirwiisp committed:
i18n: Add French translation

00-kat commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,267 @@
+#: src/apprt/gtk/ui/1.0/menu-surface-context_menu.blp:33
+#: src/apprt/gtk/ui/1.0/menu-window-titlebar_menu.blp:45
+msgid "Change Title…"
+msgstr "Changer le titre"
```
Missing ellipses, was this intentional?  (I don't know how ellipses work in French.)
```suggestion
msgstr "Changer le titre…"
```

Kirwiisp replied:
Sorry missed this one ! I'll fix this !

pluiedev commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,267 @@
+
+#: src/apprt/gtk/ui/1.5/prompt-title-dialog.blp:6
+msgid "Leave blank to restore the default title."
+msgstr "Laisser vide pour restaurer le titre par défaut"
```
Keep the original punctuation
```suggestion
msgstr "Laisser vide pour restaurer le titre par défaut."
```

pluiedev commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,267 @@
+#: src/apprt/gtk/ui/1.0/menu-surface-context_menu.blp:33
+#: src/apprt/gtk/ui/1.0/menu-window-titlebar_menu.blp:45
+msgid "Change Title…"
+msgstr "Changer le titre"
```
Keep punctuation
```suggestion
msgstr "Changer le titre…"
```

pluiedev commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,267 @@
+"clipboard contents are shown below."
+msgstr ""
+"Une application essai de lire depuis le presse-papier." 
+"Le contenu actuel du presse-papier est affiché ci-dessous"
```
```suggestion
"Le contenu actuel du presse-papier est affiché ci-dessous."
```

00-kat commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,267 @@
+msgid ""
+"One or more configuration errors were found. Please review the errors below, "
+"and either reload your configuration or ignore these errors."
+msgstr "Une ou plusieurs erreurs de configuration ont été trouvée(s). Veuillez lire les erreurs ci-dessous, et recharger votre configuration ou bien ignorer ces erreurs."
```
You might want to consider wrapping the long strings to use multiple lines similar to the source string to avoid enormous line widths.  You did it for some of the other large ones too…
```suggestion
msgstr ""
"Une ou plusieurs erreurs de configuration ont été trouvée(s). Veuillez lire les "
"erreurs ci-dessous, et recharger votre configuration ou bien ignorer ces erreurs."
```

pluiedev commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,267 @@
+"clipboard contents are shown below."
+msgstr ""
+"Une application essai d'écrire dans le presse-papier." 
+"Le contenu actuel du presse-papier est affiché ci-dessous"
```
```suggestion
"Le contenu actuel du presse-papier est affiché ci-dessous."
```

pluiedev commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,267 @@
+
+#: src/apprt/gtk/ui/1.5/ccw-paste.blp:6
+msgid "Warning: Potentially Unsafe Paste"
+msgstr "Warning: Collage potentiellement non sécurisé"
```
Is "warning" in French really also "warning"...?

Kirwiisp replied:
Yes, warning is not present in french, that's something i've tried to consider. I'll discuss with some of my friend on what might be a good translation. It's just to keep consistency between warning message we can get accross various programs. I will correct it in regard !

00-kat replied:
I'm not a native French speaker but I personally would be most used to just using `warning`, but I am biased as someone who speaks English very well; though I feel like it might be possible to pick a word which has a slightly different meaning but is close enough to work (can't recall any right now though).

> It's just to keep consistency between warning message we can get accross various programs.

Do you mean by this that other programs just use `warning` too?

00-kat replied:
> can't recall any right now though

Perhaps `attention` or `précaution` would work…

Kirwiisp replied:
I've discussed this with another french programmer. Warning seems ok to him, but like me he's used to reading and speaking english. So I've changed the translation to Attention, but might change if someone complains. I've also changed the translation to "dangereux" which might be closer to the unsafe idea.
Again, sorry for the ponctuation, I did it a bit too quickly this morning.

pluiedev commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,267 @@
+"commands may be executed."
+msgstr ""
+"Coller ce texte dans le terminal pourrait être dangereux, "
+"il semblerait que certaines commandes pourraient être exécutées"
```
```suggestion
"il semblerait que certaines commandes pourraient être exécutées."
```

pluiedev commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,267 @@
+msgid "Close Split?"
+msgstr "Fermer le panneau?"
+
+#: src/apprt/gtk/CloseDialog.zig:96
+msgid "All terminal sessions will be terminated."
+msgstr "Toutes les sessions vont être arrêtées"
+
+#: src/apprt/gtk/CloseDialog.zig:97
+msgid "All terminal sessions in this window will be terminated."
+msgstr "Toutes les sessions de cette fenêtre vont être arrêtées"
+
+#: src/apprt/gtk/CloseDialog.zig:98
+msgid "All terminal sessions in this tab will be terminated."
+msgstr "Toutes les sessions de cet onglet vont être arrêtées"
+
+#: src/apprt/gtk/CloseDialog.zig:99
+msgid "The currently running process in this split will be terminated."
+msgstr "Le processus en cours dans ce panneau va être arrêté"
```
```suggestion
#: src/apprt/gtk/CloseDialog.zig:96
msgid "All terminal sessions will be terminated."
msgstr "Toutes les sessions vont être arrêtées."

#: src/apprt/gtk/CloseDialog.zig:97
msgid "All terminal sessions in this window will be terminated."
msgstr "Toutes les sessions de cette fenêtre vont être arrêtées."

#: src/apprt/gtk/CloseDialog.zig:98
msgid "All terminal sessions in this tab will be terminated."
msgstr "Toutes les sessions de cet onglet vont être arrêtées."

#: src/apprt/gtk/CloseDialog.zig:99
msgid "The currently running process in this split will be terminated."
msgstr "Le processus en cours dans ce panneau va être arrêté."
```

Kirwiisp replied:
Sorry i didn't pay enough attention to ponctuation. I'll fix it

Kirwiisp committed:
Fix ponctuation and Warning translation

Fix: missing ponctuations on various lines.

Change "Warning" to "Attention", might change in the futur if it does not perfectly match  common practice.

Fix: Multiline on config errors dialog

Kirwiisp commented:
I've made the changes, did my best not to miss any of your feedbacks.

mitchellh commented:
@Kirwiisp I want to confirm you volunteer to maintain these?

Kirwiisp commented:
> @Kirwiisp I want to confirm you volunteer to maintain these?

Yes ! As long as the scope of the translation doesn't grow exponentially, maintaining it shouldn't be an issue !

flou commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,268 @@
+
+#: src/apprt/gtk/ui/1.5/prompt-title-dialog.blp:5
+msgid "Change Terminal Title"
+msgstr "Changer nom du terminal"
```
```suggestion
msgstr "Changer le nom du terminal"
```

flou commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,268 @@
+"An application is attempting to write to the clipboard. The current "
+"clipboard contents are shown below."
+msgstr ""
+"Une application essai d'écrire dans le presse-papier." 
```
```suggestion
"Une application essaie d'écrire dans le presse-papiers." 
```

Kirwiisp replied:
Effectively, it seems papers take the plural on Windows and MacOS, even if we can find somes spelling it singular !

flou commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,268 @@
+
+#: src/apprt/gtk/Window.zig:941
+msgid "Ghostty Developers"
+msgstr "Les developpeurs de Ghostty"
```
```suggestion
msgstr "Les développeurs de Ghostty"
```

flou commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,268 @@
+"One or more configuration errors were found. Please review the errors below, "
+"and either reload your configuration or ignore these errors."
+msgstr ""
+"Une ou plusieurs erreurs de configuration ont été trouvée(s). Veuillez lire les erreurs ci-dessous," 
```
```suggestion
"Une ou plusieurs erreurs de configuration ont été trouvées. Veuillez lire les erreurs ci-dessous," 
```

Kirwiisp replied:
I thought both were good, but it seems to take the plural of errors even if there is the possibility there is only one. It's changed ! 👍

flou commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,268 @@
+
+#: src/apprt/gtk/ui/1.5/config-errors-dialog.blp:5
+msgid "Configuration Errors"
+msgstr "Erreur de configuration"
```
```suggestion
msgstr "Erreurs de configuration"
```

flou commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,268 @@
+#: src/apprt/gtk/ui/1.5/ccw-osc-52-read.blp:6
+#: src/apprt/gtk/ui/1.5/ccw-osc-52-write.blp:6
+msgid "Authorize Clipboard Access"
+msgstr "Autoriser l'accès au presse-papier"
```
```suggestion
msgstr "Autoriser l'accès au presse-papiers"
```

flou commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,268 @@
+"An application is attempting to read from the clipboard. The current "
+"clipboard contents are shown below."
+msgstr ""
+"Une application essai de lire depuis le presse-papier." 
+"Le contenu actuel du presse-papier est affiché ci-dessous."
```
```suggestion
"Une application essai de lire depuis le presse-papiers." 
"Le contenu actuel du presse-papiers est affiché ci-dessous."
```

flou commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,268 @@
+
+#: src/apprt/gtk/Surface.zig:1243
+msgid "Copied to clipboard"
+msgstr "Copié dans le presse-papier"
```
```suggestion
msgstr "Copié dans le presse-papiers"
```

Kirwiisp committed:
fix: spelling mistakes

Pangoraw commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,268 @@
+
+#: src/apprt/gtk/CloseDialog.zig:87
+msgid "Quit Ghostty?"
+msgstr "Quitter Ghostty?"
```
```suggestion
msgstr "Quitter Ghostty ?"
```

Kirwiisp replied:
I've kept the english form, but yes in french we are used to add a space before double signs. Don't know what the prefered way for the project 🤷‍♂️

Kirwiisp replied:
@00-kat any idea on this one ? I'll change accordingly

pluiedev replied:
Always use what's most idiomatic for your language, and in this case that means space before punctuation

Kirwiisp replied:
So i've made the requested changes !

Pangoraw commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,268 @@
+
+#: src/apprt/gtk/CloseDialog.zig:88
+msgid "Close Window?"
+msgstr "Fermer la fenêtre?"
```
```suggestion
msgstr "Fermer la fenêtre ?"
```

Pangoraw commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,268 @@
+
+#: src/apprt/gtk/CloseDialog.zig:89
+msgid "Close Tab?"
+msgstr "Fermer l'onglet?"
```
```suggestion
msgstr "Fermer l'onglet ?"
```

Pangoraw commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,268 @@
+
+#: src/apprt/gtk/CloseDialog.zig:90
+msgid "Close Split?"
+msgstr "Fermer le panneau?"
```
```suggestion
msgstr "Fermer le panneau ?"
```

Pangoraw commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,268 @@
+msgid ""
+"⚠️ You're running a debug build of Ghostty! Performance will be degraded."
+msgstr ""
+"⚠️ Vous utilisez une version de débogage de Ghostty! Les performances seront dégradées."
```
```suggestion
"⚠️ Vous utilisez une version de débogage de Ghostty ! Les performances seront dégradées."
```

Kirwiisp commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,268 @@
+"An application is attempting to read from the clipboard. The current "
+"clipboard contents are shown below."
+msgstr ""
+"Une application essai de lire depuis le presse-papiers." 
```
missed this one

Kirwiisp committed:
fix: spelling and typo

flou commented on po/fr_FR.UTF-8.po:
```diff
@@ -0,0 +1,268 @@
+"clipboard contents are shown below."
+msgstr ""
+"Une application essaie d'écrire dans le presse-papiers." 
+"Le contenu actuel du presse-papier est affiché ci-dessous."
```
```suggestion
"Le contenu actuel du presse-papiers est affiché ci-dessous."
```

pluiedev reviewed (approved):
Je ne parle français pas, but I figure I should remove my previous review so that this can be merged pending bureaucratic actions from Mitchell.

Please first resolve the merge conflict, though. :)

Kirwiisp committed:
Merge branch 'main' into main

mitchellh committed:
CODEOWNERS

00-kat commented:
Hi @flou and @Pangoraw: it appears everyone forgot to ask, but would you like to join the `fr_FR` locale team?
